fix(allow-scripts): usability improvements/fixes (#1941)
LavaMoat/LavaMoat: packages/allow-scripts/README.md

In the following steps you will continue using `yarn` or `npx --no-install` if you've installed locally. Local installation is also directly callable in your project's `"scripts"` in `package.json`

Example

```sh
yarn allow-scripts setup
npx --no-install allow-scripts setup
```

> > **Warning** if @lavamoat/allow-scripts was not installed prior, npx will try to download and run allow-scripts (note no namespace prefix) which is a different package. We suggest adding --no-install to prevent accidents.

Reviewer: ```suggestion
> [!WARNING]
>
> If `@lavamoat/allow-scripts` is not already installed, `npx` will prompt to download and run package `allow-scripts` (missing the `@lavamoat` scope) **which is a different package**. We suggest using `npx --no-install` to prevent accidents.
```

allow-scripts setup
```

> Adds a `.yarnrc` or `.npmrc` (the latter if `package-lock.json` is present) to the package, populates this file with the line `ignore-scripts true`.

Reviewer: ```suggestion
Adds (or updates) a `.yarnrc` or `.npmrc` (the latter if `package-lock.json` is present) file in the current package and appends `ignore-scripts=true`.
```

> If allowScripts has an item set to true, the lifecycle scripts of that package will be executed in the run phase if the current installed version is still matching.

Reviewer: ```suggestion
If `allowScripts` has an item set to true, the lifecycle scripts of that package will be executed in the run phase if the currently installed version matches.
```

> **Note** allow-scripts is pinning versions of the dependencies with lifecycle since v5. Items set to false will get automatically updated to the current version detected. To avoid churn in the list, you can remove the `#version` part from a disallowed item and it will be denied gardless of version. Versions are mandatory for allowed packages in case of maintainer compromise. You can opt out of versions with the `--skip-versions` flag.

> > **Note** While you can configure all install scripts that you've been running to date as allowed, it's best to limit the number of them in case a package with pre-existing install script gets exploited. To figure out which packages' scripts can be ignored, try [can-i-ignore-scripts](https://www.npmjs.com/package/can-i-ignore-scripts)

Reviewer: ```suggestion
> [!NOTE]
>
> - `allow-scripts` is pinning versions of the dependencies with lifecycle since v5. 
> - Items set to `false` will be updated via `allow-scripts auto` to the current  detected version.
> - To avoid churn in the list, you can remove the `#version` suffix from a disallowed item and it will be denied regardless of version. 
> - Versions are mandatory for allowed packages in case of maintainer compromise. > - You can opt out of this behavior with the `--skip-versions` flag.

> [!NOTE]
>
> While you can configure all install scripts that you've been running to date as allowed, it's best to limit the number of them in case a package with pre-existing install script gets exploited. To determine which packages' scripts can be safely ignored, try [can-i-ignore-scripts](https://www.npmjs.com/package/can-i-ignore-scripts).
```

It will fail if it detects dependencies which haven't been set up during [configuration](#Configure) of the package. You will be advised to run `yarn allow-scripts auto`.

> ### List / debug

Reviewer: So was `debug` the old command?  Or was `list` the old command? 

```suggestion
### List

> _Alias: **debug**_
```

> Prints the complete set of information that allow-scripts uses to populate and run the allowlist content.

Reviewer: ```suggestion
Prints all information `allow-scripts` uses to populate and run the allowed scripts.
```
